curator-x-discovery/src/main/java/org/apache/curator/x/discovery/details/ServiceCacheImpl.java

```diff
@@ -66,6 +68,18 @@
         cache.getListenable().addListener(this);
     }
 
+    ServiceCacheImpl(ServiceDiscoveryImpl<T> discovery, String name, CloseableExecutorService executorService)
```

Comment:
Will do. I had to add a conversion from `ThreadFactory` to `CloseableExecutorService` in a static method since Java doesn't permit ctor's to have code before the call to the common ctor.